src/main/java/de/doubleslash/keeptime/view/ProjectReport.java

```diff
@@ -41,14 +41,22 @@ public ProjectReport(final int size) {
 
    public void appendToWorkNotes(final String currentWorkNote) {
       this.numberOfNotes++;
+      this.sb.append(currentWorkNote.trim());
       if (!currentWorkNote.isEmpty()) {
-         if (this.numberOfNotes > 1) {
+         if (this.size>1) {
             this.sb.append(NOTE_DELIMETER);
          }
-         this.sb.append(currentWorkNote.trim());
       } else {
          LOG.debug("Skipping empty note.");
       }
+
+      if(numberOfNotes==this.size){
+         int lastSemiColonIndex = this.sb.lastIndexOf(";");
+         if (lastSemiColonIndex != -1) {
```

Comment:
this logic seems really complex. I am also wondering how this works with the "lastIndexOf" and "deleteCharAt" "setLength" methods. quite confusing (for not adding a ; at the start).

As we have nice tests already I suggest to redo the whole logic here. Using class StringJoiner instead of StringBuilder ;) (I found that by accident. I know there exists some helper function to join a list of strings with a specified delimiter - but found this class which does the same thing basically.

Reply:
I changed the logic it is much simpler now and works.

Reply:
Can I delete this getter ?    
public int getNumberOfNotes() {
      return this.numberOfNotes;
I it is not used and we do not include the numberOfNotes anymore

Reply:
if not needed we can remove it :)